### PR TITLE
[OT171-36] Lastest News

### PR DESCRIPTION
**Pull request by marianobattaglia**
Creation of "Latest news" title and collection view. The images selected are the same of the last ticket so as not upload new ones that will not be used anymore when integration is done. Button at the end of CollectionView prints a string on console.

https://user-images.githubusercontent.com/94753551/161404282-a6a62503-f541-4fb4-8725-9a44889da2d4.mov

JIRA Ticket: [OT171-36]
https://alkemy-labs.atlassian.net/browse/OT171-36?atlOrigin=eyJpIjoiZTI1MzMyYmY1NzJiNGJhM2FhNDczZmQ2ZmU5YmJjNWEiLCJwIjoiaiJ9

Update 5/3/22: The requested corrections were made. I'm waiting for OT717-39 merge to pull and resolve conflicts to make the PR.

### CODE REVIEW

File: SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/NewsCollectionViewCell.swift [hunk 1]
```diff
@@ -11,11 +11,16 @@ class NewsCollectionViewCell: UICollectionViewCell {
...
-    
+    @IBOutlet weak var newsButton: UIButton!
     
     override func awakeFromNib() {
         super.awakeFromNib()
         newsDescription.font = UIFont.boldSystemFont(ofSize: 20)
+        newsButton.isHidden = true
     }
 
+    @IBAction func newsButtonTapped(_ sender: Any) {
```
Comment (gapoj) [inline SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/NewsCollectionViewCell.swift:22]: why are you putting a button in the cell of the collectionview.  this is not a propper way to capture de selection of a cell delete thisa and never do such a thing again.  Please refer to the documentation of uicollectionviewcell and its delegates,

File: SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/HomeViewController.swift [hunk 2]
```diff
@@ -2,30 +2,148 @@
...
+            cell?.myTitle.text = sliderData[indexPath.row].title
+            cell?.myImage.image = sliderData[indexPath.row].image
+            cell?.myDescription.text = sliderData[indexPath.row].description
 
-        // Do any additional setup after loading the view.
+            return cell ?? HomeCollectionViewCell()
+        } else if collectionView == self.testimoniosCollectionView {
+            let cell2 = testimoniosCollectionView.dequeueReusableCell(withReuseIdentifier: "Tcell", for: indexPath) as? TestimoniosCollectionViewCell
+            
+            cell2?.testimonioImage.image = testimoniosData[indexPath.row].image
```
Comment (gapoj) [inline SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/HomeViewController.swift:111]: you should not upload code that is not yours

File: SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/HomeViewController.swift [hunk 3]
```diff
@@ -2,30 +2,148 @@
...
+            cell2?.testimonioEpigrafe.text = testimoniosData[indexPath.row].epigrafe
+            
+            return cell2 ?? TestimoniosCollectionViewCell()
+        } else if collectionView == self.lastestNewsCollectionView { // lastestnews
```
Comment (gapoj) [inline SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/HomeViewController.swift:115]: never Use if else if else if is very bad coding, you should use a switch or something

File: SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/TestimoniosCollectionViewCell.swift [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+class TestimoniosCollectionViewCell: UICollectionViewCell {
+
+    @IBOutlet weak var testimonioImage: UIImageView!
+    @IBOutlet weak var testimonioEpigrafe: UILabel!
+    
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        // Initialization code
+    }
+
```
Comment (gapoj) [inline SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/TestimoniosCollectionViewCell.swift:19]: this not belongs to this PR

File: SomosMasApp/SomosMasApp/App/Shared/Modules/Home/View/TestimoniosCollectionViewCell.xib [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            <connections>
+                <outlet property="testimonioEpigrafe" destination="BrB-o0-znW" id="P5r-Om-SJl"/>
+                <outlet property="testimonioImage" destination="fcL-rM-x0F" id="bFr-IX-rlW"/>
+            </connections>
+            <point key="canvasLocation" x="286.95652173913044" y="140.95982142857142"/>
+        </collectionViewCell>
+    </objects>
+    <resources>
+        <systemColor name="systemBackgroundColor">
+            <color white="1" alpha="1" colorSpace="custom" customColorSpace="genericGamma22GrayColorSpace"/>
```
Comment (gapoj) [inline SomosMasApp/SomosMasApp/App/Shared/Modules/Home/View/TestimoniosCollectionViewCell.xib:69]: this not belongs to this PR

**Review by gapoj — CHANGES_REQUESTED** [review]
*the PR is not mergeable, please bring the latest from main at this point you should never ask for a PR that is not mergeable, specially if there wasn't any merge after you open the PR
* There are lot of files that you should be here because there are not yours and there are not in the main
* the level of the code that belongs to here is not acceptable, you are using buttons for things that should not, you didn't read the apples documentation about collection views or you didn't look at them in the right way...... clue instead of a button you should use the selection of the cell, very similar to the way you should do it in a tableview......
* Never use a if else if else if is be kind with your team Write readable code for example in the case use switches

**Comment by gapoj** [comment]
one last thing .... you should have the PR with all the acceptance criteria if you haven't got them don't do a PR

File: SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/HomeViewController.swift (hunk 2 above)
Comment (marianobattaglia) [reply SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/HomeViewController.swift]: Corrected, I pulled the last changes of OT171-39 with switch and rework from there.

File: SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/HomeViewController.swift (hunk 3 above)
Comment (marianobattaglia) [reply SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/HomeViewController.swift]: I code that from another branch that has a if else. Now I pulled the last changes of OT171-39 with switch and rework from there.

File: SomosMasApp/SomosMasApp/App/Shared/Modules/Home/View/TestimoniosCollectionViewCell.xib (hunk 5 above)
Comment (marianobattaglia) [reply SomosMasApp/SomosMasApp/App/Shared/Modules/Home/View/TestimoniosCollectionViewCell.xib]: File deleted from this PR.

File: SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/TestimoniosCollectionViewCell.swift (hunk 4 above)
Comment (marianobattaglia) [reply SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/TestimoniosCollectionViewCell.swift]: File deleted from this PR.

File: SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/HomeViewController.swift [hunk 6]
```diff
@@ -68,7 +86,9 @@ extension HomeViewController: UICollectionViewDataSource, UICollectionViewDelega
         case self.testimonialsCollectionView:
             // Return Max pages = 4 and add 1 more for item "Ver más"
             return min(testimonialsData.count + 1, 5)
-        // Here can be added news cases
+        case self.lastestNewsCollectionView:
+            // Return Max pages = 4 and add 1 more for item "Ver más". Its another case becouse takes data from another Array
+            return min(lastestNewsData.count + 1, 5)
```
Comment (gapoj) [inline SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/HomeViewController.swift:91]: repetitive code,  but in this case is short if you have time make it in a separate function

**Review by gapoj — COMMENTED** [review]
small corrections

File: SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/NewsCollectionViewCell.swift [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+//
+
+import UIKit
+
+class NewsCollectionViewCell: UICollectionViewCell {
+
+    @IBOutlet weak var newsImage: UIImageView!
+    @IBOutlet weak var newsDescription: UILabel!
+    
+    override func awakeFromNib() {
```
Comment (gapoj) [inline SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/NewsCollectionViewCell.swift:15]: you don´t really change nothing so you should not override here

**Review by gapoj — APPROVED** [review]

**Pull request merged**